Reorganize documentation to be more modular and cleanly grouped

choldgraf opened the pull request:
I was looking into where information for the mybinder.org cloud credit information could go, and found that it was a bit hard for me to figure out the "natural" grouping of information. We have:

- `About our team`, which as a list of team members below the fold, as well as information about team structure
- `Team resources`, which is a combination of team policy, information, and helpful guides
- `Governance`, which is about decision-making, but also includes a `milestones` section even though these aren't really "binding" from a decision-making standpoint
- `Contribute`, which is more like a general contribution guide for newcomers

So, this PR tries to do a few things with the goal of making information easier to find, more naturally grouped, and easier to decide where to update or add new information. Here's a summary:

- Our `About the team` page is now *just* a list of our current team members, and it links to a `Team information` parent page for information about the team structure.
- Our `Team resources` page is now broken down into sub-sections. One for `Governance and structure`, one for `Team guides`, that describes some common processes and policy we follow, and one for `Team coordination`, which has information about accounts and shared infrastructure + planning.
	- I created a dedicated page about `mybinder.org` infrastructure, since that's a special-case within the team. 
- I've moved `Governance` to the `Team resources` page, so we have our policy all defined in one place.
- I've explicitly labeled `Contribute` as `New contributors guide` so that it's clear what kind of information belongs there (we might consider moving this somewhere else depending on #572 )
- Also some housekeeping: I've updated `conf.py` to use the latest pydata configuration, and also removed a bunch of `sphinx-quickstart` cruft so that it is simpler

I don't believe any of the _content_ has changed, this is just rearranging

### Questions I have

- Is it actually helpful to collapse "governance" into "team information"?
- Should we instead have a separate section for "governance and policy" and differentiate this from "team information and guides?"
- Should the contributing guide be combined into two sub-sections, one for the team and one for other contributors?

choldgraf commented:
Another ping for @consideRatio since I'm on a roll w/ documentation today :-D

sgibson91 reviewed (approved):
One nitpick over word choice but shouldn't be a blocker. (Which I have just realised that I forgot to add... 🤦🏻‍♀️ )

sgibson91 commented on docs/index.rst:
```diff
@@ -27,15 +27,13 @@ governance and planning.
-course for project activity. Our overriding goal is continuous team and
-project improvement.
+This repository helps the JupyterHub team and Binder team align on our structure, policy, and goals.
+Our overriding goal is continuous team and project improvement.
```
I feel a bit nitpicky about "overriding" here. Maybe "overarching" is a better word?

I feel "overriding" is an opinionated action, like when you override a default value in code, it's because you know you want to do something different and discard whatever the software thinks you might want to do. Whereas "overarching" makes me think of many different actions that can contribute to the ultimate goal. Am I making sense?

consideRatio commented on docs/building-blocks/index.rst:
```diff
@@ -1,8 +1,8 @@
-========================
-Standard Building Blocks
-========================
+==================================
+Common workflows and helpful tools
+==================================
```
The title is updated, but not the body or folder name, which makes the title seem a bit off.

Maybe this change can be left out or followed through a bit for consistecy? Happy to see adjustment of the terminology of "standard building blocks", but I think its relevant to convey that we want to define a few ways of doing things in our jupyterhub org github repos centrally here. Currently with this suggestion, I think the `helpful` wording is a bit off. I figure `helpful tools` is like a neutral advice, while `recommended` is a clear suggestion, and `standard` is a prescription of how to do something. I see this section as at least a recommendation, possibly as a standard currently, and that is why the `helpful` adjective seems a bit off.

I figure that this text could be integrated under docs/team/repository-code-standards.md in some way in the future, or perhaps be de-nested from docs//team and become docs/repository-code-standards.

choldgraf replied:
This makes me feel like we should cleanly separate *team policy / standards* from *team recommendations / guides*. Let me try re-organizing the docs in that fashion and see how that looks. I think that'll mean re-moving the "Team contributor guide" to its own section

consideRatio commented on docs/conf.py:
```diff
@@ -1,109 +1,34 @@
+project = "Team Compass"
+copyright = "2022, JupyterHub Team and Binder Team"
+author = "JupyterHub Team and Binder Team"
+version = "1.0"
+release = "1.0"
```
I think there aren't used, and if we don't have a clear version control, we should simply not set them instead. They are documented as something you can reference from the text, but I'm not 100% if they are consumed by RTD and the UI parts they add to the website, but I don't think so.

```suggestion
```

choldgraf reviewed (commented):
Thanks for the feedback y'all - will take another quick pass and resolve these

choldgraf commented:
OK I just took another pass after @consideRatio's feedback, and made a few other significant changes:

- Top-level sections now cleanly delineate what are **Team policies and practices** from **What are team guides and information**. I hope this makes it clearer that are *expectations* vs. just helpful information.
- I've renamed and moved a couple of files / folders to make this easier to navigate and maintain. There is now a `practices/` folder (for expected practicies/standards) and a `resources/` folder (for helpful information).
- I've consolidated and removed a few redundant sections so it's clearer what information is the "source of truth" / what should be updated.

consideRatio commented on docs/practices/communication.md:
```diff
@@ -0,0 +1,45 @@
+transition easier for both developers and users, we have plugged into the repositories
+of the JupyterHub organization a [support bot](https://github.com/jupyterhub/.github/blob/master/.github/support.yml).
+
+---
+Currently, the bot will only check the following repositories:
+* [binderhub](https://github.com/jupyterhub/binderhub)
+* [jupyterhub](https://github.com/jupyterhub/jupyterhub)
+* [the-littlest-jupyterhub](https://github.com/jupyterhub/the-littlest-jupyterhub)
+---
```
These kinds of lists are unsustainable to maintain overall I think, where its better to link to the configuration where one can see what repositories its active for.

No change requested though as its not unproblematic in this case to not be mistaken about the current state sometime in the future when this is outdated.

choldgraf replied:
I agree w you. If there's a place where this is configured I'm happy to link it in this PR. If it's more complex then let's open an issue for a future fix?

consideRatio replied:
I was thinking it was you who had written these lines now, but it was just relocated - let's not add an action point for such a trivial thing!

consideRatio commented:
I'll go ahead and push some commits as I wanted to ensure that existing links redirect instead of break, so I'll be adding the sphinx extension rediraffe.

consideRatio reviewed (approved):
I added three commits, this LGTM to me.

@choldgraf can you review the added commits?

Also, could I be added to https://readthedocs.org/projects/jupyterhub-team-compass/ as well?

consideRatio commented on docs/conf.py:
```diff
@@ -1,109 +1,34 @@
 html_theme_options = {
     "use_edit_page_button": True,
-    "google_analytics_id": "UA-101904940-3",
+    "analytics": {
+        "google_analytics_id": "UA-101904940-3",
+    },
```
I forgot! Doing `make html` this showed up.

```
preparing documents... WARNING: unsupported theme option 'analytics' given
```

Looking at https://github.com/pydata/pydata-sphinx-theme/pull/650/files it seems like it should remain as it were, where `google_analytics_id` is written out directly under html_theme_options.

```suggestion
    "google_analytics_id": "UA-101904940-3",
```

consideRatio replied:
Or hmmm... This indicates we should do it like you suggested.
https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/analytics.html

consideRatio replied:
I had an outdated version of the theme installed.

choldgraf commented:
Thanks - the redirects look great to me. Also I appreciate you added the date for them, so we can decide if we want to delete them after some time has passed.

I added you as a maintainer for this repo on RTD.

consideRatio commented:
Wieeee!!! Great improvements!
